### PR TITLE
vli: vl105 onwards

### PR DESCRIPTION
**Pull request by memily**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

Per Luxshare request, adding in VL105 and onward. Testing in progress. Thanks!

### CODE REVIEW

**Comment by hughsie** [comment]
@memily does it compile locally? :)

File: plugins/vli/fu-vli-pd-device.c [hunk 1]
```diff
@@ -371,6 +371,7 @@ fu_vli_pd_device_setup(FuDevice *device, GError **error)
 						       FU_VLI_DEVICE_KIND_VL104);
 				break;
 			case 0xA0:
+				printf("vl105 found\n");
```
Comment (hughsie) [inline plugins/vli/fu-vli-pd-device.c:374]: I don't think you want this debug line

Comment (memily) [reply plugins/vli/fu-vli-pd-device.c:374]: Thanks for the catch! Would it be easier after corrections to redo the branch for a cleaner commit history?

Comment (hughsie) [reply plugins/vli/fu-vli-pd-device.c:374]: nah, just pile on the commits and we can squash it when all the CI is okay.

File: plugins/vli/fu-vli-pd-device.c [hunk 2]
```diff
@@ -348,36 +349,74 @@ fu_vli_pd_device_setup(FuDevice *device, GError **error)
...
+					    FWUPD_ERROR_INVALID_FILE,
+					    "unable to map 0x0018=0x%02X to device kind",
+					    value);
+				return FALSE;
+			}
+		} else {
+			if (!fu_vli_pd_device_read_reg(self, 0x009C, &value, error))
+				return FALSE;
+			if (!fu_vli_pd_device_read_reg(self, 0x009D, &cmp, error))
+				return FALSE;
```
Comment (hughsie) [inline plugins/vli/fu-vli-pd-device.c:397]: do these have names? I think a `#define FU_VLI_PD_REGISTER_ADDRESS_FUBAR` is much nicer to read than `0x009D`.

File: plugins/vli/fu-vli-pd-device.c [hunk 3]
```diff
@@ -629,8 +669,16 @@ fu_vli_pd_device_detach(FuDevice *device, FuProgress *progress, GError **error)
 	if (!fu_vli_pd_device_write_gpios(self, error))
 		return FALSE;
 
+	/* disable charging */
+	if (!fu_vli_pd_device_read_reg(self, 0x0003, &tmp, error))
```
Comment (hughsie) [inline plugins/vli/fu-vli-pd-device.c:673]: same, does 0x0003 have a _name_?

File: plugins/vli/fu-vli-pd-device.c (hunk 2 above)
Comment (SherlockChu219) [reply plugins/vli/fu-vli-pd-device.c:397]: How about
```
#define FU_VLI_PD_REGISTER_ADDRESS_PROJ_ID_HIGH   0x009C
#define FU_VLI_PD_REGISTER_ADDRESS_PROJ_ID_LOW   0x009D
```

Comment (hughsie) [reply plugins/vli/fu-vli-pd-device.c:397]: works for me, thanks.

File: plugins/vli/fu-vli-pd-device.c (hunk 3 above)
Comment (SherlockChu219) [reply plugins/vli/fu-vli-pd-device.c]: How about

`#define FU_VLI_PD_REGISTER_ADDRESS_GPIO_CTL_A   0x0003` 

where CTL is control?

Comment (hughsie) [reply plugins/vli/fu-vli-pd-device.c]: Maybe CONTROL_ instead? then that's great.

Comment (SherlockChu219) [reply plugins/vli/fu-vli-pd-device.c]: Noted! I'll make the changes and push -Emily

File: plugins/vli/fu-vli-pd-device.c [hunk 4]
```diff
@@ -348,36 +349,80 @@ fu_vli_pd_device_setup(FuDevice *device, GError **error)
...
-		case 0x90:
-			fu_vli_device_set_kind(FU_VLI_DEVICE(self), FU_VLI_DEVICE_KIND_VL104);
-			break;
-		default:
-			g_set_error(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_INVALID_FILE,
-				    "unable to map 0x0018=0x%02X to device kind",
-				    tmp);
+		if (!fu_vli_pd_device_read_reg(self, 0x0018, &value, error))
```
Comment (hughsie) [inline plugins/vli/fu-vli-pd-device.c:352]: what's 0x0018?

File: plugins/vli/vli-lenovo.quirk [hunk 5]
```diff
@@ -258,6 +258,29 @@ ParentGuid = USB\VID_17EF&PID_A392
 Plugin = vli
 GType = FuVliUsbhubDevice
 
+#Lenovo Test Device 2023
```
Comment (hughsie) [inline plugins/vli/vli-lenovo.quirk:261]: 100% a nit, but space after the # please

File: plugins/vli/fu-vli-pd-device.c (hunk 4 above)
Comment (memily) [reply plugins/vli/fu-vli-pd-device.c:352]: Sherlock will check.

File: plugins/vli/vli-lenovo.quirk (hunk 5 above)
Comment (memily) [reply plugins/vli/vli-lenovo.quirk:261]: No problem! I'm actually waiting on an official name for the device, please stand by.

File: plugins/vli/fu-vli-pd-device.c (hunk 4 above)
Comment (SherlockChu219) [reply plugins/vli/fu-vli-pd-device.c:352]: Can I use
`#define FU_VLI_PD_REGISTER_ADDRESS_PROJ_LEGACY   0x0018` ?
(0x0018 is old addrss of project ID, 0x009C 0x009D are new)

Comment (hughsie) [reply plugins/vli/fu-vli-pd-device.c:352]: works for me!

**Pull request merged**